src/test/IKImageTransformation.test.js

```diff
@@ -0,0 +1,135 @@
+    const ikImage = shallow(
+      <IKImage urlEndpoint={urlEndpoint} path={path} transformation={[{
+        height: 300,
+        width: 400
+      }]} transformationPosition="path" />
+    );
+    // <img alt="" src="http://ik.imagekit.io/test_imagekit_id/tr:h-300,w-400/default-image.jpg
+    // ?ik-sdk-version=react-1.0.7">
+
+    const transformURL = 'http://ik.imagekit.io/test_imagekit_id/tr:h-300,w-400/default-image.jpg';
+    expect(ikImage.find('img').prop('src')).toEqual(`${transformURL}?${global.SDK_VERSION}`);
+  });
+
+  test('transformation position as path while using absolute image path', () => {
+    const ikImage = shallow(
+      <IKImage urlEndpoint={urlEndpoint} src={src} transformation={[{
+        height: 300,
+        width: 400
+      }]} transformationPosition="path" />
+    );
+    // <img alt="" src="http://ik.imagekit.io/test_imagekit_id/default-image.jpg
+    // ?ik-sdk-version=react-1.0.7&tr=h-300%2Cw-400">
+
+    const transformURL = 'http://ik.imagekit.io/test_imagekit_id/default-image.jpg';
+    const queryParams = 'tr=h-300%2Cw-400';
+    expect(ikImage.find('img').prop('src')).toEqual(`${transformURL}?${global.SDK_VERSION}&${queryParams}`);
+  });
+
+  test('chained transformations', () => {
+    const ikImage = shallow(
+      <IKImage urlEndpoint={urlEndpoint} path={path} transformation={[{
+        height: 300,
+        width: 400
+      }, {
+        'rotation': 90
+      }]} />
+    );
+    // <img alt="" src="http://ik.imagekit.io/test_imagekit_id/tr:h-300,w-400:rt-90/default-image.jpg
+    // ?ik-sdk-version=react-1.0.7">
+
+    const transformURL = 'http://ik.imagekit.io/test_imagekit_id/tr:h-300,w-400:rt-90/default-image.jpg';
+    expect(ikImage.find('img').prop('src')).toEqual(`${transformURL}?${global.SDK_VERSION}`);
+  });
+
+  test('non-existent transformation', () => {
+    const ikImage = shallow(
+      <IKImage urlEndpoint={urlEndpoint} path={path} transformation={[{
+        'foo': 'bar',
+      }]} />
+    );
+    // <img alt="" src="http://ik.imagekit.io/test_imagekit_id/tr:foo-bar/default-image.jpg
+    // ?ik-sdk-version=react-1.0.7">
+
+    const transformURL = 'http://ik.imagekit.io/test_imagekit_id/tr:foo-bar/default-image.jpg';
+    expect(ikImage.find('img').prop('src')).toEqual(`${transformURL}?${global.SDK_VERSION}`);
+  });
+
+  test('non-existent transformation with existing transformation', () => {
+    const ikImage = shallow(
+      <IKImage urlEndpoint={urlEndpoint} path={path} transformation={[{
+        'foo': 'bar',
+        height: 300
+      }]} />
+    );
+    // <img alt="" src="http://ik.imagekit.io/test_imagekit_id/tr:foo-bar,h-300/default-image.jpg
+    // ?ik-sdk-version=react-1.0.7">
+
+    const transformURL = 'http://ik.imagekit.io/test_imagekit_id/tr:foo-bar,h-300/default-image.jpg';
+    expect(ikImage.find('img').prop('src')).toEqual(`${transformURL}?${global.SDK_VERSION}`);
+  });
+});
+
+describe('IKImage transformations', () => {
```

Comment:
The group name `IKImage transformations` is not clarifying what it consists of. It should be clear what each test case/group is doing just by looking at the output.

```
PASS src/test/IKUpload.test.js
  IKUpload
    Success case
      ✓ should successfully upload file (54 ms)
    Failure cases
      ✓ should call onError for missing publicKey (4 ms)
      ✓ should call onError for missing urlEndpoint (2 ms)
      ✓ should call onError for missing authenticationEndpoint (3 ms)
      ✓ should call onError in case server returns error response (4 ms)
    General cases
      ✓ should call onChange when file input changes (3 ms)
  IKUpload snapshots
    ✓ with authentication (2 ms)
    ✓ with all the props (2 ms)
    ✓ without fileName prop (1 ms)
    ✓ without authentication (2 ms)

PASS src/test/IKImage.test.js
  IKImage snapshots
    ✓ path with transformations and lqip (2 ms)
    ✓ nested path with lqip (3 ms)
    absolute image path
      ✓ src with alt attribute (7 ms)
      ✓ src with query parameters (2 ms)
      ✓ src with transformation (1 ms)
      ✓ src with lqip (3 ms)
      ✓ src with transformation and lqip (2 ms)
    relative image path
      ✓ path with alt attribute (3 ms)
      ✓ path with query parameters (1 ms)
      ✓ path having leading slashes (2 ms)
      ✓ path with url endpoint having trailing slashes (2 ms)
      ✓ path with lqip (3 ms)
      ✓ path with transformations (2 ms)

PASS src/test/IKImageTransformation.test.js
  IKImage transformation snapshots
    ✓ single transformation (3 ms)
    ✓ transformation position as query (2 ms)
    ✓ transformation position as path while using relative image path (2 ms)
    ✓ transformation position as path while using absolute image path (1 ms)
    ✓ chained transformations (2 ms)
    ✓ non-existent transformation (2 ms)
    ✓ non-existent transformation with existing transformation (1 ms)
  IKImage transformations
    ✓ observer disconnects when component unmounts (3 ms)

PASS src/test/IKContext.test.js
  IKContext snapshots
    ✓ with only urlEndpoint (19 ms)
    ✓ with publicKey and urlEndpoint (3 ms)
    ✓ with publicKey, urlEndpoint and authenticationEndpoint (4 ms)
    ✓ overriding urlEndpoint (4 ms)

PASS src/test/ErrorBoundary.test.js
  ErrorBoundary
    ✓ handles error thrown by child component (21 ms)
  ErrorBoundary snapshots
    IKImage
      ✓ missingUrlEndpointFail (70 ms)
      ✓ imageUrlEndpointFail (4 ms)
```

Reply:
Grouping and test case labeling improved for all components.